src/cryptoadvance/specter/server_endpoints/settings.py

```diff
@@ -117,6 +117,7 @@ def general():
                         continue
                 write_wallet(wallet)
                 app.specter.wallet_manager.update()
+                time.sleep(1)
```

Comment:
During restoration process, after selecting the files I was getting the error that the wallets linked with the hot wallet were not restored because they do not exist. So after seeing logs I concluded that before wallet manager could update all the wallets, following line was being executed and giving the error. So to avoid that this sleep is needed.
`wallet_obj = app.specter.wallet_manager.get_by_alias(wallet["alias"])`

Reply:
I think a better approach here would be to not use the threading in `wallet_manager.update()` you could add to that update funtion a parameter like `threading=True`, then in line 105 of the wallet_manager do:
```
if self.allow_threading and threading:
```

Or something like that...

Reply:
Updated in latest commit